src/aave-v2/SupplyVaultBase.sol

```diff
@@ -39,33 +35,40 @@ abstract contract SupplyVaultBase is ISupplyVaultBase, ERC4626UpgradeableSafe, O
     /// @notice Thrown when the zero address is passed as input or is the recipient address when calling `transferRewards`.
     error ZeroAddress();
 
-    /// CONSTANTS AND IMMUTABLES ///
+    /// IMMUTABLES ///
 
     IMorpho public immutable morpho; // The main Morpho contract.
     ERC20 public immutable morphoToken; // The address of the Morpho Token.
     ILens public immutable lens; // The address of the Morpho Lens.
+    address public immutable recipient; // The recipient of the rewards that will redistribute them to vault's users.
 
     /// STORAGE ///
 
     address public poolToken; // The pool token corresponding to the market to supply to through this vault.
-    address public recipient; // The recipient of the rewards that will redistribute them to vault's users.
 
     /// CONSTRUCTOR ///
 
     /// @dev Initializes network-wide immutables.
     /// @param _morpho The address of the main Morpho contract.
     /// @param _morphoToken The address of the Morpho Token.
     /// @param _lens The address of the Morpho Lens.
+    /// @param _recipient The recipient of the rewards that will redistribute them to vault's users.
     constructor(
         address _morpho,
         address _morphoToken,
-        address _lens
+        address _lens,
+        address _recipient
     ) {
-        if (_morpho == address(0) || _morphoToken == address(0) || _lens == address(0))
-            revert ZeroAddress();
+        if (
+            _morpho == address(0) ||
+            _morphoToken == address(0) ||
+            _lens == address(0) ||
+            _recipient == address(0)
```

Comment:
Because `recipient` is an immutable, one could argue that if we want to disable transferring rewards, we could just upgrade and remove this condition (or just the rewards transfer, whatever), enabling us to effectively disable rewards transfer

On the other hand, enabling this now would allow us to trustlessly re-deploy the same implementation, with a different `recipient` value (in this case, `address(0)`)

Considering the unlikelihood of disabling the transfer of rewards, I'm ok with this change (and the removal of the `address(0)` check)!

Reply:
>enabling us to effectively disable rewards transfer

not exactly since the address 0 check has been removed from the `transferRewards` function